skymp5-server/cpp/server_guest_lib/script_classes/PapyrusUtility.cpp

```diff
@@ -57,14 +40,193 @@ VarValue PapyrusUtility::GetCurrentRealTime(
+  VarValue self, const std::vector<VarValue>& arguments)
+{
+  return ArrayHelper(self, "ResizeAliasArray", arguments,
+                     VarValue::kType_ObjectArray, true);
+}
+
+VarValue PapyrusUtility::ResizeBoolArray(
+  VarValue self, const std::vector<VarValue>& arguments)
+{
+  return ArrayHelper(self, "ResizeBoolArray", arguments,
+                     VarValue::kType_BoolArray, true);
+}
+
+VarValue PapyrusUtility::ResizeFloatArray(
+  VarValue self, const std::vector<VarValue>& arguments)
+{
+  return ArrayHelper(self, "ResizeFloatArray", arguments,
+                     VarValue::kType_FloatArray, true);
+}
+
+VarValue PapyrusUtility::ResizeFormArray(
+  VarValue self, const std::vector<VarValue>& arguments)
+{
+  return ArrayHelper(self, "ResizeFormArray", arguments,
+                     VarValue::kType_ObjectArray, true);
+}
+
+VarValue PapyrusUtility::ResizeIntArray(VarValue self,
+                                        const std::vector<VarValue>& arguments)
+{
+  return ArrayHelper(self, "ResizeIntArray", arguments,
+                     VarValue::kType_IntArray, true);
+}
+
 void PapyrusUtility::Register(
   VirtualMachine& vm, std::shared_ptr<IPapyrusCompatibilityPolicy> policy)
 
 {
   compatibilityPolicy = policy;
 
   AddStatic(vm, "Wait", &PapyrusUtility::Wait);
+  AddStatic(vm, "WaitMenuMode", &PapyrusUtility::WaitMenuMode);
+  AddStatic(vm, "WaitGameTime", &PapyrusUtility::WaitGameTime);
+  AddStatic(vm, "IsInMenuMode", &PapyrusUtility::IsInMenuMode);
   AddStatic(vm, "RandomInt", &PapyrusUtility::RandomInt);
   AddStatic(vm, "RandomFloat", &PapyrusUtility::RandomFloat);
   AddStatic(vm, "GetCurrentRealTime", &PapyrusUtility::GetCurrentRealTime);
+  AddStatic(vm, "GetCurrentGameTime", &PapyrusUtility::GetCurrentGameTime);
+  AddStatic(vm, "GameTimeToString", &PapyrusUtility::GameTimeToString);
+  AddStatic(vm, "CreateAliasArray", &PapyrusUtility::CreateAliasArray);
+  AddStatic(vm, "CreateBoolArray", &PapyrusUtility::CreateBoolArray);
+  AddStatic(vm, "CreateFloatArray", &PapyrusUtility::CreateFloatArray);
+  AddStatic(vm, "CreateFormArray", &PapyrusUtility::CreateFormArray);
+  AddStatic(vm, "CreateIntArray", &PapyrusUtility::CreateIntArray);
+  AddStatic(vm, "ResizeAliasArray", &PapyrusUtility::ResizeAliasArray);
+  AddStatic(vm, "ResizeBoolArray", &PapyrusUtility::ResizeBoolArray);
+  AddStatic(vm, "ResizeFloatArray", &PapyrusUtility::ResizeFloatArray);
+  AddStatic(vm, "ResizeFormArray", &PapyrusUtility::ResizeFormArray);
+  AddStatic(vm, "ResizeIntArray", &PapyrusUtility::ResizeIntArray);
+}
+
+VarValue PapyrusUtility::WaitHelper(VarValue& self, const char* funcName,
+                                    const std::vector<VarValue>& arguments)
+{
+  if (arguments.size() < 1)
+    throw std::runtime_error(std::string(funcName) +
+                             " requires at least 1 argument");
+  double seconds = static_cast<double>(arguments[0].CastToFloat());
+  if (0.0 >= seconds)
```

Comment:
Incompatible change, will break `while (1) Wait(0)`

Reply:
If return without promise breaks something then we shouldn't do it. Changed.